code/controllers/subsystem/codex.dm

```diff
@@ -168,3 +241,120 @@ SUBSYSTEM_DEF(codex)
 		if(entry)
 			present_codex_entry(showing_mob, entry)
 			return TRUE
+
+#define CODEX_SEARCH_INDEX_FILE "codex.db"
+#define CODEX_SERIAL_ALWAYS_VALID "I_DOWNLOADED_A_ZIP_INSTEAD_OF_USING_GIT"
+/// Prepare the search database.
+/datum/controller/subsystem/codex/proc/prepare_search_database(drop_existing = FALSE)
+	if(GLOB.is_debug_server && !FORCE_CODEX_DATABASE)
+		to_chat(world, span_debug("Codex: Debug server detected. DB operation disabled."))
+		return
+	if(drop_existing)
+		to_chat(world, span_debug("Codex: Deleting old index..."))
+		//Check if we've already opened one this round, if so, get rid of it.
+		if(codex_index)
+			del(codex_index)
+		fdel(CODEX_SEARCH_INDEX_FILE)
+	else
+		to_chat(world, span_debug("Codex: Preparing Search Database"))
+
+
+	if(!rustg_file_exists(CODEX_SEARCH_INDEX_FILE))
+		if(!drop_existing)
+			to_chat(world, span_debug("Codex: Database missing, building..."))
+		create_db()
+		build_db_index()
+
+	if(!codex_index) //If we didn't just create it, we need to load it.
+		codex_index = new(CODEX_SEARCH_INDEX_FILE)
+
+	var/database/query/cursor = new("SELECT * FROM _info")
+	if(!cursor.Execute(codex_index))
+		to_chat(world, span_debug("Codex: ABORTING! Database error: [cursor.Error()] | [cursor.ErrorMsg()]"))
+		return
+
+	cursor.NextRow()
+	var/list/revline = cursor.GetRowData()
+	var/db_serial = revline["revision"]
+	if(db_serial != GLOB.revdata.commit)
+		if(db_serial == CODEX_SERIAL_ALWAYS_VALID)
+			to_chat(world, span_debug("Codex: Special Database Serial detected. Data may be inaccurate or out of date."))
+		else
+			to_chat(world, span_debug("Codex: Database out of date, Rebuilding..."))
+			prepare_search_database(TRUE) //recursiveness funny,,
+			return
+
+	if(drop_existing)
+		to_chat(world, span_debug("Codex: Collation complete.\nCodex: Index ready."))
+		return
+	to_chat(world, span_debug("Codex: Database Serial validated.\nCodex: Loading complete."))
+
+/datum/controller/subsystem/codex/proc/create_db()
+	// No index? Make one.
+
+	to_chat(world, span_debug("Codex: Writing new database file..."))
+	//We explicitly store the DB in the root directory, so that TGS builds wipe it.
+	codex_index = new(CODEX_SEARCH_INDEX_FILE)
+
+	//Create the initial schema
+	var/database/query/init_cursor = new({"
+/* Holds the revision the index was compiled for. If it's different then live, we need to regenerate the index.*/
+	CREATE TABLE "_info" (
+		"revision"	TEXT
+	);
+	"})
```

Comment:
is this a comment in the database itself?
also write this string in a seperate var and pass it to `new`, easier to read.